### ISSUE TITLE
tus-upload stdin pipe

### ISSUE THREAD

**Issue by anacrolix** [cenk/tus.py#2]
Piping to tus-upload doesn't appear to work: If I give the file as /dev/stdin, I get the error:
```
Traceback (most recent call last):
  File "/usr/local/bin/tus-upload", line 11, in <module>
    sys.exit(_cmd_upload())
  File "/usr/local/lib/python2.7/dist-packages/tus.py", line 83, in _cmd_upload
    file_size = _get_file_size(args.file)
  File "/usr/local/lib/python2.7/dist-packages/tus.py", line 142, in _get_file_size
    pos = f.tell()
IOError: [Errno 29] Illegal seek
```
It would seem this is supported by the protocol however, with Upload-Defer-Length. Note that if I give /dev/stdin, and specify the file directly with input redirection `<`, tus-upload is able to seek on the file descriptor and determine the size.

**Comment by cenkalti** [comment]
Hello @anacrolix, I have fixed the problem you described in version [1.3.2](https://pypi.org/project/tus.py/1.3.2/).

Now the following command works correctly:
```cat $FILE | tus-upload /dev/stdin $TUS_BASE```

However, I couldn't handle resuming for streams. I guess it is not trivial to support resuming for streams. Do you have a use case?

**Comment by anacrolix** [comment]
I'll give this a try. My use case is streaming a download straight to another destination without storing locally. I'd guess resume could be achieved by maybe storing each chunk in a temporary file until it succeeds, or buffering a certain amount ahead of the upload position and discarding as chunks succeed.
